src/analysis/SmartDetectorsAnalysis/SmartDetectorRunner.cs

```diff
@@ -130,11 +135,44 @@ public async Task<List<ContractsAlert>> RunAsync(SmartDetectorAnalysisRequest re
                 }
 
                 // Create results
+                bool detectorSupportsAutomaticResolution = smartDetector is IAutomaticResolutionSmartDetector;
                 List<ContractsAlert> results = new List<ContractsAlert>();
                 foreach (var alert in alerts)
                 {
                     QueryRunInfo queryRunInfo = await this.queryRunInfoProvider.GetQueryRunInfoAsync(new List<ResourceIdentifier>() { alert.ResourceIdentifier }, cancellationToken);
-                    results.Add(alert.CreateContractsAlert(request, smartDetectorManifest.Name, queryRunInfo, this.analysisServicesFactory.UsedLogAnalysisClient, this.analysisServicesFactory.UsedMetricClient));
+                    ContractsAlert contractsAlert = alert.CreateContractsAlert(
+                        request,
+                        smartDetectorManifest.Name,
+                        queryRunInfo,
+                        this.analysisServicesFactory.UsedLogAnalysisClient,
+                        this.analysisServicesFactory.UsedMetricClient);
+
+                    // Handle automatic resolution parameters in the alerts:
+                    // If the detector supports automatic resolution - save the predicates for the resolution checks
+                    // If the detector supports automatic resolution - drop the automatic resolution parameters (since they are useless) and error trace
+                    if (contractsAlert.AutomaticResolutionParameters != null)
+                    {
+                        if (detectorSupportsAutomaticResolution)
+                        {
+                            this.tracer.TraceInformation($"Alert {contractsAlert.Id} has automatic resolution parameters, so saving alert details for later use");
+                            await stateRepository.StoreStateAsync(
```

Comment:
Should we prevent the user from accessing or deleting this state?

Reply:
we can't realistically do it as the developer has access to the application's resources
i don't think there's a need for added logic here

---
In reply to: [234988292](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r234988292) [](ancestors = 234988292)